servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/activateMembership.js

```diff
@@ -0,0 +1,65 @@
+const { Roles } = require('@orbiting/backend-modules-auth')
+const activateMembership = require('../../../lib/activateMembership')
+const { publishMonitor } = require('../../../../../lib/slack')
+
+module.exports = async (_, args, { pgdb, req, t, user: me, mail: { enforceSubscriptions } }) => {
+  Roles.ensureUserHasRole(me, 'supporter')
+
+  const { id: membershipId } = args
+
+  let activatedMembership
+  const transaction = await pgdb.transactionBegin()
+
+  try {
+    const periods = await transaction.public.membershipPeriods.find({ membershipId })
+    if (periods.length) {
+      throw new Error(t('api/membership/activate/hasPeriods'))
+    }
+
+    const membership = await transaction.public.memberships.findOne({ id: membershipId })
+    if (!membership) {
+      throw new Error(t('api/membership/activate/404'))
+    }
+
+    if (membership.active) {
+      throw new Error(t('api/membership/activate/alreadyActive'))
+    }
+
+    const user = await transaction.public.users.findOne({ id: membership.userId })
+    if (!user) {
+      throw new Error(t('api/membership/activate/userMissing'))
+    }
+
+    activatedMembership = await activateMembership(membership, user, t, pgdb)
+
```

Comment:
```suggestion
    await transaction.transactionCommit()

```

Reply:
I would close the transaction try-catch here and do `enforceSubscriptions` and `publishMonitor` outside of it